plugins/module_utils/network/sonic/config/bgp_neighbors_af/bgp_neighbors_af.py

```diff
@@ -219,13 +244,77 @@ def set_state(self, want, have):
         diff = get_diff(want, have, TEST_KEYS)
 
         if state == 'overridden':
-            commands, requests = self._state_overridden(want, have, diff)
+            commands, requests = self._state_overridden(want, have)
         elif state == 'deleted':
-            commands, requests = self._state_deleted(want, have, diff)
+            commands, requests = self._state_deleted(want, have)
         elif state == 'merged':
             commands, requests = self._state_merged(want, have, diff)
         elif state == 'replaced':
-            commands, requests = self._state_replaced(want, have, diff)
+            commands, requests = self._state_replaced(want, have)
+        return commands, requests
+
+    def _state_replaced(self, want, have):
+        """ The command generator when state is replaced
+
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        commands, requests = [], []
+        self.sort_lists_in_config(want)
+        self.sort_lists_in_config(have)
+        new_have = deepcopy(have)
+        new_want = deepcopy(want)
+        for default_entry in default_entries:
+            remove_matching_defaults(new_have, default_entry)
+            remove_matching_defaults(new_want, default_entry)
+        new_have = remove_empties_from_list(new_have)
+        new_want = remove_empties_from_list(new_want)
+        add_config, del_config = self._get_replaced_config(new_want, new_have)
+        if del_config:
+            del_requests = self.get_delete_bgp_neighbors_af_requests(del_config, new_have, (del_config == new_have))
```

Comment:
"del_config" and "new_have" must be sorted to do proper comparison. Can the line264 and 265 be moved at before line#275 and sort "del_config" and "new_have"?

Reply:
Removed sort_lists_in_config in all the codes.

Reply:
You may need sort_lists_in_config for "del_config" and "new_have" for proper comparison.

Reply:
Added the sorting in del_config and new_have.